test_files/addon_test.py

```diff
@@ -33,6 +42,63 @@ def test_disable_enable(self):
         bpy.ops.preferences.addon_disable(module="MCprep_addon")
         bpy.ops.preferences.addon_enable(module="MCprep_addon")
 
+    def test_translations(self):
+        """Safely ensures translations are working fine still.
+
+        Something go wrong, blender stuck in another language? Run in console:
+        bpy.context.preferences.view.language = "en_US"
+        """
+        init_lang = bpy.context.preferences.view.language
+        try:
+            self._test_translation()
+        except Exception:
+            raise
+        finally:
+            bpy.context.preferences.view.language = init_lang
+
+    def _test_translation(self):
+        """Ensure that creating the MCprep environment is error-free."""
+        test_env = conf.MCprepEnv()
+        mcprep_dir = self.addon_path()
+        lang_folder = mcprep_dir / "MCprep_resources" / "Languages"
+        test_env.languages_folder = lang_folder
+
+        print("Lang folder", lang_folder)
+        print("self.translations", test_env.use_direct_i18n)
+        print("Dir:", list(os.listdir(lang_folder)))
+
+        # Don't assign translations, to force building the map in memory
+        # translations_py = mcprep_dir / "translations.py"
+        # test_env.translations = translations_py
+
+        # Force load translations into this instance of MCprepEnv
+        test_env._load_translations(prefix=">>>\t")
+
+        # TODO: Not working, above load encounters error
+        self.assertIn(
+            "en_US", test_env.languages, "Missing default translation key")
+        self.assertTrue(
+            test_env.use_direct_i18n, "use_direct_i18n should be True")
+
+        # Magic string evaluations, will break if source po's change
+        test_translations = [
+            ("ru_RU", "Restart blender", "Перезапустите блендер"),
+            ("zh_HANS", "Texture pack folder", "材质包文件夹"),
+            ("en_US", "Mob Spawner", "Mob Spawner"),
+        ]
+        for lang, src, dst in test_translations:
+            with self.subTest(lang):
+                # First ensure the mo files exist
+                self.assertTrue(
+                    os.path.isfile(
+                        lang_folder / lang / "LC_MESSAGES" / "mcprep.mo"),
+                    f"Missing {lang}'s mo file")
```

Comment:
If nothing else, we could reduce the test down to just this, which already goes a long way to prove that translations are present when they should be.